Source/Task/AsyncLib.cpp

```diff
@@ -430,6 +430,8 @@ static HRESULT AllocStateNoCompletion(_Inout_ XAsyncBlock* asyncBlock, _Inout_ A
 
 static HRESULT AllocState(_Inout_ XAsyncBlock* asyncBlock, _In_ size_t contextSize)
 {
+    RETURN_HR_IF(E_INVALIDARG, asyncBlock == nullptr);
```

Comment:
In the GDK code guidelines we are asked to dereference null args instead of returning an error.  The idea is the resulting crash is much easier to diagnose than weeding down where an E_INVALIDARG came from.